docs/development/contributor_guidelines.rst

```diff
@@ -107,6 +107,82 @@ Run the checks locally via:
 
       make yamllint
 
+Type hints in Python code
+-------------------------
+
```

Comment:
I realize this kind of out-of-scope of the docs but do you think you could add a very short statement of `why we decided to add type annotation to SD` ? I think it will help new-comers who might be kind of confused before they start digging through the docs.

Reply:
Perfect, will do so.